pkg/apis/acme/v1/types_order.go

```diff
@@ -74,13 +74,15 @@ type OrderSpec struct {
 	// DNSNames is a list of DNS names that should be included as part of the Order
 	// validation process.
 	// This field must match the corresponding field on the DER encoded CSR.
-	//+optional
+	// +optional
+	// +listType=set
 	DNSNames []string `json:"dnsNames,omitempty"`
 
 	// IPAddresses is a list of IP addresses that should be included as part of the Order
 	// validation process.
 	// This field must match the corresponding field on the DER encoded CSR.
 	// +optional
+	// +listType=set
```

Comment:
```suggestion
	// +listType=atomic
```
Since "This field must match the corresponding field on the DER encoded CSR.", it should be owned by a single controller that also owns the Request field.

Reply:
Agreed, reverted.